engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compiledgraph/evaluatedparam.scala

```diff
@@ -1,14 +1,26 @@
 package pl.touk.nussknacker.engine.compiledgraph
 
-import pl.touk.nussknacker.engine.api.expression.{Expression, ExpressionTypingInfo, TypedValue}
+import pl.touk.nussknacker.engine.api.definition
+import pl.touk.nussknacker.engine.api.expression.{Expression, ExpressionTypingInfo, TypedExpression, TypedValue}
 import pl.touk.nussknacker.engine.api.typed.typing.TypingResult
 
 object evaluatedparam {
 
   case class TypedParameter(name: String, typedValue: TypedValue)
 
-  case class Parameter(name: String, expression: Expression, returnType: TypingResult,
-                       shouldBeWrappedWithScalaOption: Boolean, shouldBeWrappedWithJavaOptional: Boolean,
-                       typingInfo: ExpressionTypingInfo)
+  case class Parameter(typedExpression: TypedExpression, parameterDefinition: definition.Parameter) {
```

Comment:
Won't will cause some serialization problems? WDYT about replacing it with `apply()` method instead?

Reply:
:+1: I added apply instead